### PR TITLE
poker - Update tests to match canonical-data.json @v1.0.0

### LINKED ISSUE #1002: poker: check test version and update if necessary
As identified in #784, some of the tests currently don't reference the version number for the `canonical-data.json` that they are based on, `poker` is one of them. In order to help with the ongoing maintenance of this track, we're in the process of checking the tests against the current version of the `canonical-data.json` - if they are up-to-date, then you can just add the version string, but if they are out-of-date, then the tests will need updating.

Tests should typically be ordered according to their order in the canonical data, since this makes it easier to check and update them in future. If you think that there's a better order, please let us know your reasoning in your PR.

If there are some additional tests that currently exist in this track but which aren't in the canonical data, these can be kept if you think that they are useful. They should be placed at the bottom of the test cases, and it would also be super-helpful if you could add a comment before them saying `# Additional tests for this track` to make it clear that they aren't part of the canonical data.

The version string should be placed just before the first test class and followed by a blank line. For this exercise, the version string is:
```
# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0
```

**Test suite:** [exercism/python/exercises/poker/poker_test.py](https://github.com/exercism/python/blob/master/exercises/poker/poker_test.py)
**Canonical data:** [exercism/problem-specifications/exercises/poker/canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/poker/canonical-data.json)

### PR DESCRIPTION
**Pull request by njgingrich**
Resolves #1002.
Updated tests to match canonical data.

I renamed some tests to better match up with the json file, and some of the implementations had slightly different cards so those were matched.

### CODE REVIEW

File: exercises/poker/poker_test.py [hunk 1]
```diff
@@ -2,118 +2,159 @@
 
 from poker import poker
 
+# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0
```
Comment (cmccandless) [inline exercises/poker/poker_test.py]: Can you please add an additional line above this (2 above, 1 below) for consistency with the track?

**Comment by njgingrich** [comment]
Sure thing! Sorry about the delay.

**Comment by cmccandless** [comment]
@njgingrich Merged. Congrats! Thanks for working on this!

**Comment by njgingrich** [comment]
Thanks!

**Pull request merged**